pkg/scheduler/framework/plugins/tainttoleration/taint_toleration.go

```diff
@@ -61,15 +61,18 @@ func (pl *TaintToleration) Filter(ctx context.Context, state *framework.CycleSta
 	if err != nil {
 		return framework.NewStatus(framework.Error, err.Error())
 	}
+	var taintNotTolerated *v1.Taint
```

Comment:
Either
a) delete this, make the if use a `:=`, and move the error construction into an `else` block, OR
b) set this variable here, not inside the if block.

Setting the variable in the if expression makes the most sense when the variable's scope is confined within the if/else blocks.

Reply:
I vote b

Reply:
`taintNotTolerated` using it's value outside of the if block, in making the `errReason`